pykafka/utils/compression.py

```diff
@@ -79,15 +79,22 @@ def encode_snappy(buff, xerial_compatible=False, xerial_blocksize=32 * 1024):
     Adapted from kafka-python
     https://github.com/mumrah/kafka-python/pull/127/files
     """
+    buff = get_bytes(buff)
+    #snappy segfaults if it gets a read-only buffer on PyPy
+    if IS_PYPY:
```

Comment:
I thought about providing this inside the buffer compat but it's only a bug for snappy so it seemed most relevant here